andela/Ah-backend-guardians
# #162685679 Highlight and comment on any text in an article

**What does this PR do?**
This PR delivers a feature that enables a user to highlight and comment on any text in an article

**Description of Task to be completed?**
The task to be completed is intended to ensuring that users should be able to highlight and comment on any text in an article

**What are the relevant Pivotal Tracker stories?**
[#162685679](https://www.pivotaltracker.com/story/show/162685679)

**How should it be manually tested?**
- Clone the GitHub repository from your terminal by using `git clone https://github.com/andela/Ah-backend-guardians.git`
- cd into the folder and create a virtual environment with command `virtualenv venv`
- Activate the virtual environment using the command `. venv/bin/activate` and then  install dependencies with the command `pip install -r requirements.txt`
- Run `python manage.py makemigrations` then `python manage.py migrate`
- Run the server locally with `python manage.py runserver`
- Open postman and follow the instructions below :
- Create a user and login to the system using `http://localhost:8000/api/users/` and `http://localhost:8000/api/users/login/` respectively with POST methods
- Use the POST method with this URL `
http://localhost:8000/api/<article-slug>/comments/
` after creating an article using POST method on `
http://localhost:8000/api/articles/
`
Article body 
```
{
	"title”:”My Demo Article",
	"description":"Demo Article",
	"tags":["Demo","Lorem"],
	"body":"At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non p
}
```
```
Comment body 
{
	"body":"Comment on a portion of the article",
	"start_index":50,
	"end_index":130
}
```
Ensure to use integer value for the indices.

Results are as shown in the images below

<img width="732" alt="screen shot 2019-01-30 at 9 58 03 am" src="https://user-images.githubusercontent.com/40206038/51963826-cd8f3980-2475-11e9-895b-34becb939ff0.png">

<img width="1109" alt="screen shot 2019-01-30 at 9 58 23 am" src="https://user-images.githubusercontent.com/40206038/51963844-d4b64780-2475-11e9-8605-b73bf88d9ba6.png">
When you fetch the comments...
